File: src/main/java/com/example/dpop/orchestrator/session/IdentificationAttempt.java
package com.example.dpop.orchestrator.session;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

import java.time.Instant;
import java.util.UUID;

@Entity
@DiscriminatorValue("IDENTIFICATION")
public class IdentificationAttempt extends OrchestratorAttempt {

    protected IdentificationAttempt() {
    }

    public IdentificationAttempt(UUID processSessionId, String nextContext, String nextStep, Instant expiresAt) {
        super(processSessionId, nextContext, nextStep, expiresAt);
    }
}

File: src/main/kotlin/com/example/dpop/orchestrator/session/IdentificationAttempt.kt
package com.example.dpop.orchestrator.session

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import java.time.Instant
import java.util.UUID

@Entity
@DiscriminatorValue("identification")
class IdentificationAttempt : OrchestratorAttempt {

    protected constructor() : super()

    constructor(
        processSessionId: UUID?,
        nextContext: String?,
        nextStep: String?,
        expiresAt: Instant?
    ) : super(processSessionId, nextContext, nextStep, expiresAt)
}
